.gitlab-ci.yml

```diff
@@ -1,14 +1,78 @@
-test:
-  image: "python:3.6-stretch"
+stages:
+  - test
+  - package
+
+.install_deps: &install_system_deps
+  before_script:
+    - >
+      dnf install -y --refresh python3 python3-devel 'python3dist(pip)'
+      'python3dist(tox)' make gcc which xz xorriso libxml2-devel libxslt-devel
+      enchant genisoimage ShellCheck
+
+tox_check:
+  stage: test
+  image: fedora:latest
+  variables:
+    LANG: en_US.UTF-8
+  <<: *install_system_deps
   script:
-    - apt-get update -qq
-    - apt-get install -y git
-    - apt-get install -y xsltproc
-    - apt-get install -y genisoimage
-    - apt-get install -y enchant
-    - apt-get install -y shellcheck
-    - apt-get install -y python3-pip
-    - pip3 install --upgrade pip
-    - pip install tox
-    - tox -e check
+    - dnf install -y python36 python34 python2 python2-devel python2-virtualenv python2-pip
+    # Python 2.7
+    - export PYTHON=python2.7
+    - tox -e unit_py2_7 "-n $(nproc)"
+    # Python 3.4
+    - export PYTHON=python3.4
+    - tox -e unit_py3_4 "-n $(nproc)"
+    # Python 3.6
+    - export PYTHON=python3.6
     - tox -e unit_py3_6 "-n $(nproc)"
+    # Python 3.7
+    - export PYTHON=python3.7
+    - tox -e 'unit_py3_7,check' "-n $(nproc)"
+  cache:
+    key: "$CI_JOB_NAME"
+    paths:
+      - .tox
+
+build_doc:
+  stage: test
+  image: fedora:latest
+  variables:
+    TOXENV: 'packagedoc'
+    PYTHON: 'python3.7'
+  <<: *install_system_deps
+  script:
+    - >
+      dnf -y install latexmk texlive-cmap texlive-metafont texlive-ec
+      texlive-babel-english texlive-fncychap texlive-fancyhdr texlive-titlesec
+      texlive-tabulary texlive-framed texlive-wrapfig texlive-parskip
+      texlive-upquote texlive-capt-of texlive-needspace texlive-makeindex
+      texlive-times texlive-helvetic texlive-courier texlive-gsftopk
+      texlive-updmap-map texlive-dvips
+    - tox
+  artifacts:
+    paths:
+      - doc/build/
+  cache:
+    key: "$CI_JOB_NAME"
+    paths:
+      - .tox/
+      - doc/build/
+
+RPM:
+  stage: package
+  image: fedora:latest
+  before_script:
+    - dnf --refresh -y install mock perl-TimeDate which make gzip tar perl git
+  script:
+    - 'sed -i "s|build: clean tox|build:|" Makefile'
+    - make build
+    - mv dist/python-kiwi.spec .
+    - rm dist/python-kiwi.changes
```

Comment:
We should test the conversion of changes into a changelog... Perhaps we can do that later, but we should consider that...

Reply:
This isn't really urgent, so we can merge #997 first  and then I'll rebase this branch on master. However, I am not sure if the current RPM stage will catch any issues with the changelog, as it is using the tools from Fedora (since I don't know how to run osc without credentials).

I can run rpmlint on the result though.

Reply:
@dcermak You can install `obs-build` and do the following:

```bash
$ /usr/lib/build/changelog2spec dist/python-kiwi.changes >> dist/python-kiwi.spec
```

Reply:
@Conan-Kudo could you create an issue with your idea in this regard ? I would put that as an extra PR and continue getting this one merged. It looks really good now

Reply:
Will do.